classes-recipe/platsch.bbclass

```diff
@@ -0,0 +1,138 @@
+                # bytes. For the i.MX6/i.MX8 the bitmap needs to be flipped (see [1]). This
+                # is done via the bitbake parameter MGK_CUSTOM_OPTION (see [2]). As the BMP
+                # format stores potential color profiles behind the desired pixel array at
+                # the end of the file those particularly need to be dropped before by the
+                # option "-strip" to let the hack succeed (see [3]). Explicitly request an
+                # RGB format to avoid color map and alpha channel (via asking for the
+                # TrueColor type, see [4]).
+                #
+                # [1] https://github.com/pengutronix/meta-ptx/pull/117#issuecomment-1474786927
+                # [2] https://imagemagick.org/script/command-line-options.php#flip
+                # [3] https://imagemagick.org/script/command-line-options.php#strip
+                # [4] https://imagemagick.org/script/command-line-options.php#type
+                #
+                SIZE_OF_RAW_BITMAP=$(expr ${WIDTH} "*" ${HEIGHT} "*" "(" 5 + 6 + 5 ")" / 8)
+                env MAGICK_CONFIGURE_PATH=${RECIPE_SYSROOT_NATIVE}/etc/ImageMagick-7 magick.im7 \
+                    ${WORKDIR}/${IMG} ${MGK_CUSTOM_OPTION} \
+                    -strip -type TrueColor -define bmp:subtype=${SPLASH_FORMAT} bmp:- | \
+                    tail -c ${SIZE_OF_RAW_BITMAP} > ${B}/${BASE}-${WIDTH}x${HEIGHT}-${SPLASH_FORMAT}.bin
+                ;;
+            *)
+                bbfatal "platsch.bbclass: unknown SPLASH_FORMAT \"${SPLASH_FORMAT}\"."
+                ;;
+            esac
+
+            # Help the developer of a converter to a new target format with a good error
+            # message.
+            if [ -z "${SIZE_OF_RAW_BITMAP}" ]; then
+                bbfatal "platsch.bbclass: forgot to calculate SIZE_OF_RAW_BITMAP in above case clause for SPLASH_FORMAT \"${SPLASH_FORMAT}\"."
+            fi
+
+            # Sanity check that the size of the raw bitmap is as predicted (this e.g.
+            # spotlighted that ImageMagick per default converts color mapped bitmaps to
+            # color mapped bitmaps and thereby led to the introduction of option "-type
+            # TrueColor").
+            FILESIZE=$(stat --format="%s" -- ${B}/${BASE}-${WIDTH}x${HEIGHT}-${SPLASH_FORMAT}.bin)
+            if [ "${FILESIZE}" -ne "${SIZE_OF_RAW_BITMAP}" ]; then
+                bbfatal "platsch.bbclass: FILESIZE=${FILESIZE} while SIZE_OF_RAW_BITMAP=${SIZE_OF_RAW_BITMAP} - something is fishy in the conversion of \"${IMG}\"!"
+            fi
+        }
+    done
+}
+
+platsch_do_install() {
+    # Take care to only call 'install' if there are adequately formatted bitmaps as
+    # otherwise the second invocation of 'install' would fail. This is equivalent to
+    # the following wildcard pattern expanding to something different:
+    if [ "$(echo ${B}/*-${SPLASH_FORMAT}.bin)" != "${B}/*-${SPLASH_FORMAT}.bin" ]; then
```

Comment:
This is just to catch the corner case that there are no converted bitmaps at all which should only be possible if the list ``SPLASH_IMAGES`` is empty. But seeing that list empty is not sufficient for the conclusion that there are generally no images to be installed as images could have been add in a recipe's ``do_compile:append()`` by other means from different source files.

Reply:
Mh, if there are no images at all, would one want the recipe to ignore that explicitly? Or wouldn't an error be what should be expected?

Reply:
The intent of this check originally only was to avoid that the second install command fails and not to protect the user from accidental misuse. But you're right: why not alert the user of the class that perhaps something went unexpected? Although I would prefer to do that via ``bbwarn`` instead of ``bbfatal``. What do you think about the following update?

Reply:
Ok, still not fully convinced that the if clause is required but let's give it a try :smirk: